### ISSUE TITLE
Saving drafts with missing required fields

### ISSUE THREAD

**Issue by antonok-edm** [modrinth/labrinth#413]
#### Is your feature request related to a problem? Please describe.
I attempted to prepare and save a draft project for the first release of my mod. I planned to update some fields later on, mostly involving the actual JAR (there are still a couple more small changes I'd like to make to the mod itself), but this issue could easily apply for other required fields as well.

However, I was unable to save the draft project without creating at least one version with at least one valid JAR file. Uploading an empty file, a small text file, or an empty zip renamed as a JAR file also didn't work. It seems wasteful to upload a JAR to Modrinth that I know will not be used in practice. Not accepting empty fields in the draft also increases the likelihood that the creator will forget to update them in the future when submitting for review.

#### Describe the solution you'd like
It should be possible to save a project as a draft with required fields missing. Validation for missing fields would still occur when submitting the project for review.

#### Describe alternatives you've considered
Some other forms of validation could potentially be loosened during drafting, although _some_ validation should definitely still be performed for security and other concerns. I think allowing required fields to be empty might be the least controversial restriction to loosen for drafts.

**Comment by haykam821** [comment]
I've been able to create drafts without versions already as recently as last week. This might just be a bug or regression.

**Comment by Geometrically** [comment]
Not a bug? I just tried on the frontend and I could create a draft without versions just fine. Your case may be specific, and if it's not letitng you create a mod, send a message over in our support channels.

**Comment by antonok-edm** [comment]
Ah sorry, a mistake on my part, but still a valid issue, I believe.

You _can_ indeed create a draft without a version. I misread the alarming-looking red text in the versions section, but it _does_ mention that it's only required for submitting for review. I just tested and it works fine without any version in a new draft.


That being said, there are still a bunch of fields that _must_ to be filled in to create a draft; in particular:
```
Your project must have a name.
Your project must have a summary.
Your project must have a vanity URL.
Your project must have a license.
```

I can see why the URL slug might be necessary, and maybe name is a reasonable thing to enforce as well, but otherwise a mod author may wish to save their work before deciding on a final summary and license. Sure, they can add a random placeholder, but it increases the odds of forgetting to update it before submitting.

Also, if any version is added:
```
Your version must have a unique version number.
Your version must have the supported mod loaders selected.
Your version must have the supported Minecraft versions selected.
Your version must have a file uploaded.
```

This was the real issue for me; I didn't have the final JAR for my mod ready at draft time, but I'd already filled in everything else. I had to get around the restriction by creating a zip archive containing just a `fabric.mod.json` with an empty JSON object and renaming it to a `.jar` extension. It took me a while to get past all the file validation criteria, but I definitely didn't want to lose the rest of my work on the draft.
